skyportal/handlers/api/sharing.py

```diff
@@ -51,42 +51,76 @@ def post(self):
         spec_ids = data.get("spectrumIDs", [])
         if not phot_ids and not spec_ids:
             return self.error(
-                "One of either `photometryIDs` or `spectrumIDs` " "must be provided."
+                "One of either `photometryIDs` or `spectrumIDs` must be provided."
             )
-        groups = Group.query.filter(Group.id.in_(group_ids))
-        if not all([group in self.current_user.accessible_groups for group in groups]):
-            return self.error(
-                "Insufficient permissions: you must have access to each "
-                "target group you wish to share data with."
-            )
-        obj_id = None
+
+        ok_groups = Group.query.filter(Group.id.in_(group_ids))
+        ok_group_ids = [og.id for og in ok_groups]
+        invalid_group_ids = [gid for gid in group_ids if gid not in ok_group_ids]
+
+        if len(invalid_group_ids) > 0:
+            return self.error(f'Invalid group IDs: {invalid_group_ids}.')
+        groups = ok_groups
+
+        phot_obj_ids = []
+        spec_obj_ids = []
+
         if phot_ids:
-            query = Photometry.query.filter(Photometry.id.in_(phot_ids))
-            for phot in query:
+            ok_phot = Photometry.query.filter(Photometry.id.in_(phot_ids))
+            ok_phot_ids = [op.id for op in ok_phot]
```

Comment:
same comment here (ok -> valid)
ok feels ambiguous to me as I can't tell whether it's ok from the perspective of a valid ID or is accessible to user